wireless/bluetooth/bt_atomic.h

```diff
@@ -37,26 +37,20 @@
  * Pre-processor Definitions
  ****************************************************************************/
 
-#define bt_atomic_set(ptr, value)     (*(ptr) = (value))
-#define bt_atomic_get(ptr)            (*(ptr))
-#define bt_atomic_testbit(ptr, bitno) ((*(ptr) & (1 << (bitno))) != 0)
+#define bt_atomic_set(ptr, value)        atomic_store((ptr), (value));
+#define bt_atomic_get(ptr)               atomic_load(ptr)
+#define bt_atomic_testbit(ptr, bitno)    ((atomic_load(ptr) & (1 << (bitno))) != 0)
+#define bt_atomic_incr(ptr)              atomic_fetch_add((ptr), 1)
+#define bt_atomic_decr(ptr)              atomic_fetch_sub((ptr), 1)
+#define bt_atomic_setbit(ptr, bitno)     atomic_fetch_or((ptr), ((1 << (bitno))))
+#define bt_atomic_clrbit(ptr, bitno)     atomic_fetch_and((ptr), (~(1 << (bitno))))
+#define bt_atomic_testsetbit(ptr, bitno) ((atomic_fetch_or((ptr), (1 << (bitno))) & (1 << (bitno))) != 0)
+#define bt_atomic_testclrbit(ptr, bitno) ((atomic_fetch_and((ptr), ~(1 << (bitno))) & (1 << (bitno))) != 0)
 
 /****************************************************************************
  * Public Types
  ****************************************************************************/
 
-typedef uint8_t bt_atomic_t;
-
-/****************************************************************************
- * Public Function Prototypes
- ****************************************************************************/
-
-bt_atomic_t bt_atomic_incr(FAR bt_atomic_t *ptr);
-bt_atomic_t bt_atomic_decr(FAR bt_atomic_t *ptr);
-bt_atomic_t bt_atomic_setbit(FAR bt_atomic_t *ptr, bt_atomic_t bitno);
-bt_atomic_t bt_atomic_clrbit(FAR bt_atomic_t *ptr, bt_atomic_t bitno);
-
-bool bt_atomic_testsetbit(FAR bt_atomic_t *ptr, bt_atomic_t bitno);
-bool bt_atomic_testclrbit(FAR bt_atomic_t *ptr, bt_atomic_t bitno);
+typedef atomic_char bt_atomic_t;
```

Comment:
algin with https://github.com/apache/nuttx/pull/14827:
```suggestion
typedef atomic_t bt_atomic_t;
```